lib/videoroom/meeting.ex

```diff
@@ -0,0 +1,159 @@
+    try do
+      GenServer.call(registry_id(meeting_name), :add_peer)
+    catch
+      :exit, {:noproc, _error} ->
+        {:error, "Failed to add peer"}
+    end
+  end
+
+  # Callbacks
+
+  @impl true
+  def init(args) do
+    args = %{
+      name: Keyword.fetch!(args, :name),
+      max_peers: Keyword.get(args, :max_peers)
+    }
+
+    Logger.metadata(room_name: args.name)
+
+    client = Jellyfish.Client.new()
+
+    with {:ok, notifier} <- Jellyfish.Notifier.start(),
+         {:ok, room} <- find_or_create_room(client, args) do
+      Process.monitor(notifier)
+
+      Logger.info("Created meeting")
+
+      {:ok, %{client: client, notifier: notifier, name: args.name, room_id: room.id}}
+    else
+      {:error, reason} ->
+        raise "Failed to create a meeting, reason: #{inspect(reason)}"
+    end
+  end
+
+  @spec find_or_create_room(Jellyfish.Client.t(), %{name: binary(), max_peers: integer()}) ::
+          {:ok, Room.t()} | {:error, atom() | binary()}
+  defp find_or_create_room(client, args) do
+    with {:ok, room_id} <- RoomRegistry.lookup(args.name),
+         {:ok, room} <- Room.get(client, room_id) do
+      {:ok, room}
+    else
+      {:error, :unregistered} ->
+        create_new_room(client, args)
+
+      error ->
+        handle_room_error(error, client, args)
+    end
+  end
+
+  defp create_new_room(client, args) do
+    case Room.create(client, max_peers: args.max_peers) do
+      {:ok, room} ->
+        RoomRegistry.insert_new(args.name, room.id)
+        {:ok, room}
+
+      error ->
+        error
+    end
+  end
+
+  defp handle_room_error({:error, reason} = error, client, args) do
+    if String.contains?(reason, "does not exist") do
+      RoomRegistry.delete(args.name)
+      create_new_room(client, args)
+    else
+      error
+    end
+  end
+
+  @impl true
+  def handle_call(:add_peer, _from, state) do
+    case Room.add_peer(state.client, state.room_id, Jellyfish.Peer.WebRTC) do
+      {:ok, peer, token} ->
+        Logger.info("Added peer #{peer.id}")
+        {:reply, {:ok, token}, state}
+
+      _error ->
+        Logger.warning("Failed to add peer")
+        {:reply, {:error, "Failed to add peer"}, state}
+    end
+  end
+
+  @impl true
+  def handle_info({:jellyfish, %type{} = notification}, state)
+      when type in [PeerDisconnected, PeerCrashed, RoomCrashed] do
+    if notification.room_id == state.room_id do
+      Logger.info("jellyfish notification: #{inspect(notification)}")
+      handle_notification(notification, state)
+    else
+      Logger.debug("jellyfish notification: #{inspect(notification)}")
+      {:noreply, state}
+    end
+  end
+
+  def handle_info({:jellyfish, notification}, state) do
+    Logger.debug("jellyfish notification: #{inspect(notification)}")
+    {:noreply, state}
+  end
+
+  def handle_info({:DOWN, _ref, :process, pid, _reason}, %{notifier: notifier})
+      when pid == notifier do
+    raise "Connection to jellyfish closed!"
+  end
```

Comment:
Why not 
```suggestion
  def handle_info({:DOWN, _ref, :process, pid, _reason}, %{notifier: pid}) do
    raise "Connection to jellyfish closed!"
  end
```
We don't expect `:DOWN` from any other processes, do we?